### ISSUE TITLE
Solving with conda throws an error when the default conda path doesn't include "/env/"

### ISSUE THREAD

**Issue by tylerpotts** [Netflix/metaflow#1293]
TLDR metaflow requires the use of conda environment paths that include `/env/` in the pathname, when conda itself does not have that requirement

I'm currently managing a jupyterhub deployment where our conda environments look like this (from `conda info`):

```
       envs directories : /opt/userenvs/tyler.potts
                          /opt/envs/shared
                          /opt/conda
                          /home/tyler.potts/.conda/envs
                          /opt/conda/envs
```
We're doing this because the path with `/envs/` is on a shared AWS EFS filesystem so all users can have access to a core set of conda environments. Users have the ability to locally create custom conda environments that aren't on the EFS filesystem, which requires a different path.

When I attempt to solve a simple flow with conda such as this with `python helloflow.py --environment=conda run`:

```
from metaflow import FlowSpec, step, conda_base

libraries = {
    "pip": "23.0.1"
}
@conda_base(python="3.11.0", libraries=libraries)
class HelloConda(FlowSpec):

    @step
    def start(self):
        self.next(self.end)

    @step
    def end(self):
        print("HelloFlow is all done.")

if __name__ == "__main__":
    HelloConda()
```

The following error is thrown:

```
  File "/opt/userenvs/tyler.potts/metaflow/lib/python3.11/site-packages/metaflow/plugins/conda/conda.py", line 156, in _deps
    for package in self.package_info(env_id):
  File "/opt/userenvs/tyler.potts/metaflow/lib/python3.11/site-packages/metaflow/plugins/conda/conda.py", line 105, in package_info
    metadata = os.path.join(self._env_path(env_id), "conda-meta")
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 76, in join
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

This is because [here](https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/conda/conda.py#L96) and [here](https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/conda/conda.py#L167) the conda plugin is filtering out environment paths that don't contain `/env/` and the function returns `None`. This `None` object is passed along until it creates the error above.

I've created a PR to fix this here: #1294
